sdk/packages/selector-sui/src/adapter.ts

```diff
@@ -0,0 +1,495 @@
+      return true
+    }
+
+    if (
+      this._eagerConnectForStandardWallets &&
+      getRecentStandardWalletForNetwork(SUI_NETWORK) !== null &&
+      isStandardConnectedForNetwork(SUI_NETWORK)
+    ) {
+      return true
+    }
+
+    return false
+  }
+
+  eagerConnectDeeplink = () => {
+    if (isMobileBrowser() && this._app) {
+      const mobileWalletName = getRecentStandardWalletForNetwork(SUI_NETWORK)
+      const wallet = this._walletsList.find((w) => w.name === mobileWalletName)
+
+      if (typeof wallet === 'undefined') {
+        return
+      }
+
+      if (wallet.deeplink === null) {
+        return
+      }
+
+      if (wallet.deeplink.universal !== null) {
+        this._app.connectDeeplink({
+          walletName: wallet.name,
+          url: wallet.deeplink.universal
+        })
+        return
+      }
+
+      if (wallet.deeplink.native !== null) {
+        this._app.connectDeeplink({
+          walletName: wallet.name,
+          url: wallet.deeplink.native
+        })
+      }
+    }
+  }
+
+  connectToMobileWallet = (walletName: string) => {
+    if (this._modal) {
+      this._modal.setStandardWalletConnectProgress(true)
+    }
+
+    const wallet = this._walletsList.find((w) => w.name === walletName)
+
+    if (!this._app || typeof wallet === 'undefined') {
+      return
+    }
+
+    if (wallet.deeplink === null) {
+      return
+    }
+
+    if (wallet.deeplink.universal !== null) {
+      this._app.connectDeeplink({
+        walletName: wallet.name,
+        url: wallet.deeplink.universal
+      })
+
+      this._chosenMobileWalletName = walletName
+
+      triggerConnect(
+        wallet.deeplink.universal,
+        this._app.sessionId,
+        this._appInitData.url ?? 'https://nc2.nightly.app'
+      )
+      return
+    }
+
+    if (wallet.deeplink.native !== null) {
+      this._app.connectDeeplink({
+        walletName: wallet.name,
+        url: wallet.deeplink.native
+      })
+
+      this._chosenMobileWalletName = walletName
+
+      triggerConnect(
+        wallet.deeplink.native,
+        this._app.sessionId,
+        this._appInitData.url ?? 'https://nc2.nightly.app'
+      )
+    }
+  }
+  connectToStandardWallet = async (walletName: string, onSuccess: () => void) => {
+    if (this._modal) {
+      this._modal.setStandardWalletConnectProgress(true)
+    }
+    const wallet = this._walletsList.find((w) => w.name === walletName)
+    if (typeof wallet?.standardWallet === 'undefined') {
+      return
+    }
+
+    const adapter = new StandardWalletAdapter({
+      wallet: wallet.standardWallet as StandardWalletAdapterWallet
+    })
+
+    try {
+      await adapter.connect()
+
+      persistRecentStandardWalletForNetwork(walletName, SUI_NETWORK)
+      persistStandardConnectForNetwork(SUI_NETWORK)
+      this._connectionType = ConnectionType.WalletStandard
+      this._innerStandardAdapter = adapter
+      this._accounts = (await adapter.getAccounts()).map((a) => a)
```

Comment:
is this map necessary?

Reply:
yeah since TS complains about read only access